config/routes.rb

```diff
@@ -68,4 +68,6 @@
     end
     resources :data_requests
   end
+
+  get 'my/user_reductions/:reducible_type/:reducible_id', to: 'user_reductions#current_user_reductions', format: :json
```

Comment:
As user reduction is a resource in the caesar app, i'd probably look at changing the route to specify that and use query params to modify the resource scoping of the user_reductions you want. I'd remove the `my` prefix and invert the paradigm to access scoping based on supplied id args

`/user_reductions/` # get me all user reductions (index) - admin can be all but scoped to that user
`/user_reductions/?user_id=1` # get me all the specified user reductions (index) -  scoped to the user id
`/user_reductions/?user_id=1&workflow_id=1234` # get me all my user reductions for workflow 1234
`/user_reductions/?user_id=1&project_id=4321` # get me all my user reductions forproject_id=4321
`/user_reductions/?user_id=1&workflow_id=1234&project_id=4321` # get me all my user reductions for both workflow project combos they should match or it returns nothing.

Reply:
the routes didn't end up looking like this, but they follow basically the same principle.  since user_reductions aren't a root-level resource, the information about how to filter them is already implied by the nesting :)

Reply:
`UserReduction` is a model and model resources are normally scoped on their own routes. How do you show a particular UserReduction?

I can see why you've setup the routes this way but i still think it's backwards, the resource is a UserReduction, the modifications of the wired scopes can be changed via query params.

Reply:
That's an interesting question; so far, we have no use case for showing a particular user reduction! Right now, if you want to pluck one to update, you retrieve it by the workflow or project that owns it, and I think in general the routes are trying to suggest "X owns Y" relationships with nesting (which is why I asked a rather silly question in dev-backend about whether pundit was invoked at each intermediate resource).

Reply:
@marten set up the initial routing and since then we've more or less copied him. But there is certainly a difference between extractors, which belong very concretely to a workflow, and user reductions, where the connection is a little more tenuous

Reply:
Not a blocking comment from me about your routes. 

If you don't have need to show a resource i.e. it's a filtered index route only and you always need to scope the filtering via project_id and / or workflow_id then it LGTM. 

APIs can change if you haven't got a public surface that will break clients :shipit:

Reply:
Yes, you're right. I set the routes up with nesting initially because it makes little sense to me for e.g. a project owner to request all of the "extractor" models across workflows. I think I'm in more of a DHH camp than Cam or Parish were on this front.

With extracts and reductions, I'm beginning to lean towards removing at least some of the nesting, maybe I'd keep `/projects/extracts?column=blah` but drop the nesting underneath a specific extractor key. But I really don't mind either way, it's very much a preference and it shouldn't be my preference dictating, especially now. Do whatever you prefer and I'll join you.